src/freenas/etc/grub.d/10_truenas_linux

```diff
@@ -320,7 +320,40 @@ for dataset in `echo $datasets`; do
   fi
 
   boot_device_id=
-  list="/boot/vmlinuz-$kernel_version"
+  # See if debugkernel is set to boot
+  debugkernel=`midclt call system.advanced.config | jq '.debugkernel' 2>/dev/null`
```

Comment:
This script can be ran by ISO installer, in that case middleware will not be available. You'll need to manually read the database file using sqlite3 CLI

Reply:
Thanks, I have moved to reading the database file `/data/freenas-v1.db`. I have tested it by replacing the script on running system. That is working.  Building an image to test the install and update process from the ISO.

Reply:
I have tested the update and installation in following scenarios with latest changes:

- Installing the .update file from WebUI.
- Installing the update by booting the ISO and upgrading an existing installation and preserving the previous BE.
- Fresh install on a system, destroying the previous installations and formatting the boot device.

In all the scenarios, update/install is successful. We are able to boot into previous and new installations from GRUB after install/update is successful. All the boot entries are functioning correctly and boot the system with selected entry. if debugkernel is set to true and `update-grub` is run, system boots with debug kernel on next boot from the entry that is selected bu default present on main menu of GRUB.